ts/WoltLabSuite/Core/Ui/Moderation/Clipboard/AssignUser.ts

```diff
@@ -0,0 +1,187 @@
+import UiDialog from "../../Dialog";
+
+interface EventData {
+  data: ClipboardActionData;
+  listItem: HTMLLIElement;
+  responseData: AjaxResponse;
+}
+
+class UiModerationClipboardAssignUser implements AjaxCallbackObject, DialogCallbackObject {
+  /**
+   * ids of the moderation queue entries currently being handled
+   */
+  protected queueIds: number[] = [];
+
+  public _ajaxFailure(data: ResponseData): boolean {
+    if (data.returnValues?.fieldName === "assignedUsername") {
+      let errorMessage = "";
+
+      const dialog = UiDialog.getDialog(this)!.content;
+      const assignedUsername = dialog.querySelector("input[name=assignedUsername]") as HTMLInputElement;
+
+      const errorType: string = data.returnValues.errorType;
+      switch (errorType) {
+        case "empty":
+          errorMessage = Language.get("wcf.global.form.error.empty");
+          break;
+
+        case "notAffected":
+          errorMessage = Language.get("wcf.moderation.assignedUser.error.notAffected");
+          break;
+
+        default:
+          errorMessage = Language.get(`wcf.user.username.error.${errorType}`, {
+            username: assignedUsername.value,
+          });
+          break;
+      }
+
+      DomUtil.innerError(assignedUsername, errorMessage);
+
+      return false;
+    }
+
+    return true;
+  }
+
+  public _ajaxSetup(): ReturnType<AjaxCallbackSetup> {
+    return {
+      data: {
+        actionName: "assignUserByClipboard",
+        className: "wcf\\data\\moderation\\queue\\ModerationQueueAction",
+      },
+    };
+  }
+
+  public _ajaxSuccess(): void {
+    UiDialog.close(this);
+
+    UiNotification.show(undefined, () => window.location.reload());
+  }
+
+  public _dialogSetup(): ReturnType<DialogCallbackSetup> {
+    const submitCallback = () => this.submitDialog();
+
+    return {
+      id: "moderationQueueClipboardAssignUser",
+      options: {
+        onSetup(content: HTMLElement): void {
+          const username = content.querySelector("input[name=assignedUsername]") as HTMLInputElement;
+          new UiUserSearchInput(username, {});
+
+          username.addEventListener("click", (event) => {
+            const assignedUserId = DomTraverse.prevBySel(
+              event.currentTarget as HTMLElement,
+              "input[name=assignedUserID]",
+            ) as HTMLInputElement;
+            assignedUserId.click();
+          });
+
+          content.querySelector("button[data-type=submit]")!.addEventListener("click", submitCallback);
+        },
+        onShow(content: HTMLElement): void {
+          // Reset dialog to initial state.
+          const assignedUsername = content.querySelector("input[name=assignedUsername]") as HTMLInputElement;
+          (content.querySelector("input[name=assignedUserID]:checked") as HTMLInputElement).checked = false;
+          (content.querySelector("input[name=assignedUserID]") as HTMLInputElement).checked = true;
```

Comment:
Aren't these equivalent to simply setting `.checked = true` for all inputs with name `assignedUserID`?

Reply:
`assignedUserID` are radiobuttons.

Reply:
Ah, yeah. And it's `.querySelector`, so line 108 will match the first element only. I find this logic pretty hard to understand. Are you able to work with `defaultChecked` here? I never tested this, but something like:

```
content.querySelectorAll("input[name=assignedUserID]").forEach((el) => el.checked = el.defaultChecked)
```

could work based off the documentation.

Reply:
Yes, it works.